lightning_server.cc

```diff
@@ -77,6 +77,9 @@ void LightningServer::start() {
     // Routing requests to their handlers
     RequestHandler* handler = router.routeRequest(request->uri());
     RequestHandler::Status status_code = handler->handleRequest(*request, &response);
+    if (status_code == RequestHandler::NOT_FOUND) {
```

Comment:
For documentation: the logic error here is that if the StaticFileHandler is invoked (because we're visiting `/static/filename.txt`, but the file does not exist, the NotFoundHandler is never invoked. That's because we return the NotFoundHandler for when a route does not match, but we forgot to invoke it for when a route does match, but we subsequently hit another error.

Very nice catch, @apfong!